target/consensource-ui
# Factory search bugfixes

## Proposed change/fix

Implement table filtering. The URL search params are kept in sync with the table state and serve as a means of populating filters on initial load.

## Types of changes

What types of changes does this pull request introduce to ConsenSource? _Put an `x` in the boxes that apply_


## How to run/test

- Follow setup instructions from https://github.com/target/consensource-ui/pull/56. You will need at least two factories, so create another one through the UI.
- Navigate to `localhost:3000/search`
- Perform the following actions and verify that the UI is updated accordingly. Note that not all of the backend filters are implemented, so the records returned will not be in alignment with the active filters
  - Address filtering
    - Open the search bar
    - Type `hello world`
    - Confirm that the query params are `?address=hello%20world`
    -  Refresh the page, and confirm the search filter is still populated
  - Custom text field filtering (e.g. `name` param)
    - Open the filters tab and fill in the `name` value. 
    - Press `Apply Filters`
    - Confirm there is a filter chip with the `Name: ` key, and the query params are `?name=<val>`
    - Refresh the page and confirm the filter chip appears, and that the filter for `name` is populated in the filter modal
    - Click the `X` on the filter chip
    - Confirm that both the chip are removed and the query param is removed.
  - Custom multiselect (e.g. `countries` param)
    - Open the filters tab and select two countries from the `Countries` multiselect
    - Press `Apply Filters`
    - Confirm there are two filter chips, and that the query params are `?country=<country1>,<country2>`
    - Refresh the page
    - Confirm both filter chips appear, and that the values are checked in the `Countries` multiselect`
    - Remove a single filter chip
    - Confirm both the query params, filter chip, and filter modal are updated.
